### ISSUE TITLE
Enhancement: Caching for characters (Mid Song Character Switching)

### ISSUE THREAD

**Issue by radpas1213** [FunkinCrew/Funkin#3055]
### Issue Checklist

- [X] I have properly named the enhancement
- [X] I have checked the issues/discussions pages to see if the enhancement has been previously suggested

### What is your suggestion, and why should it be implemented?

Currently characters only load the sprite sheet when it is created, this is not good for songs that have mid song character changes.
If there is a function for each CharacterRenderType to cache the sprite sheets before the stage loads it would be very beneficial.

**Comment by radpas1213** [comment]
Yeah, I'm making a port of Mortis and right now I've decided to make the characters that are a bit too heavy when switching into extra characters instead.

**Comment by Cartridge-Man** [comment]
Well then this issue doesn’t matter, does it?

**Comment by Cartridge-Man** [comment]
Because what I’m trying to say is that the event is made by a modder and therefore it isn’t a problem with the base game. So either modify the event yourself or ask the creator of the event to make it better. Because I’m pretty sure the base game doesn’t have this

**Comment by Cartridge-Man** [comment]
So close this issue, is what I’m trying to say if my message wasn’t already incredibly transparent and the point of it so visible.

**Comment by AbnormalPoof** [comment]
Anyways, for an actual helpful answer: FunkinSprite has a `cacheSparrow` function that allows you to preload sprites, reducing lag when they're shown on screen. Alternatively, you can also simply hide the character (setting the alpha to an extremely low value) when the countdown starts, and then quickly swap them out when the event is triggered.

**Comment by Cartridge-Man** [comment]
> Anyways, for an actual helpful answer:

ouch.

**Comment by Starexify** [comment]
Mod issue ? 
The game can't fix your code...
<sub>You have to cache and handle the characters YOURSELF if you want that</sub>
